clients/include/testing_csrmm.hpp

```diff
@@ -447,45 +431,43 @@ hipsparseStatus_t testing_csrmm(Arguments argus)
         }
         else
         {
-            gen_matrix_coo(nrows, ncols, nnz, hcoo_row_indA, hcsr_col_indA, hcsr_valA, idx_base);
+            gen_matrix_coo(M, K, nnz, hcoo_row_indA, hcsr_col_indA, hcsr_valA, idx_base);
         }
 
         // Convert COO to CSR
-        hcsr_row_ptrA.resize(nrows + 1, 0);
+        hcsr_row_ptrA.resize(M + 1, 0);
         for(int i = 0; i < nnz; ++i)
         {
             ++hcsr_row_ptrA[hcoo_row_indA[i] + 1 - idx_base];
         }
 
         hcsr_row_ptrA[0] = idx_base;
-        for(int i = 0; i < nrows; ++i)
+        for(int i = 0; i < M; ++i)
         {
             hcsr_row_ptrA[i + 1] += hcsr_row_ptrA[i];
         }
     }
 
-    M = (transA == HIPSPARSE_OPERATION_NON_TRANSPOSE ? nrows : ncols);
-    K = (transA == HIPSPARSE_OPERATION_NON_TRANSPOSE ? ncols : nrows);
-
     // Some matrix properties
-    int A_m = (transA == HIPSPARSE_OPERATION_NON_TRANSPOSE ? M : K);
-    int A_n = (transA == HIPSPARSE_OPERATION_NON_TRANSPOSE ? K : M);
-    int B_m = (transB == HIPSPARSE_OPERATION_NON_TRANSPOSE ? K : N);
-    int B_n = (transB == HIPSPARSE_OPERATION_NON_TRANSPOSE ? N : K);
-    int C_m = M;
-    int C_n = N;
-    ldb     = order == HIPSPARSE_ORDER_COLUMN
-                  ? (transB == HIPSPARSE_OPERATION_NON_TRANSPOSE ? 2 * K : 2 * N)
-                  : (transB == HIPSPARSE_OPERATION_NON_TRANSPOSE ? 2 * N : 2 * K);
-    ldc     = order == HIPSPARSE_ORDER_COLUMN ? 2 * M : 2 * N;
-
-    int nrowB = order == HIPSPARSE_ORDER_COLUMN ? ldb : B_m;
-    int ncolB = order == HIPSPARSE_ORDER_COLUMN ? B_n : ldb;
-    int nrowC = order == HIPSPARSE_ORDER_COLUMN ? ldc : C_m;
-    int ncolC = order == HIPSPARSE_ORDER_COLUMN ? C_n : ldc;
-
-    int Bnnz = nrowB * ncolB;
-    int Cnnz = nrowC * ncolC;
+    int A_m = M;
```

Comment:
In csrmm, A is always mxk (unlike in spmm_csr where op(A) is mxk)

Reply:
? in spmm_csr op(A) is mxk?..

Reply:
Yes I think so. From the cusparse documentation for spmm: 

"where op(A) is a sparse matrix with dimensions m × k "

I think this is what your PR was fixing in csrmm because I mistakenly thought they both followed this but its only spmm and not csrmm?

Reply:
This is why your PR only needed to change testing_csrmm and not testing_spmm_csr right?

Reply:
From cusparse documentation for csrmm:

"A is an m×k sparse matrix that is defined in CSR..."

Reply:
If my understanding is wrong though please correct me. I assumed that is was the bug that your PR was fixing

Reply:
I think A is always `m x k`, you are passing A to the function, not op(A).

Reply:
mmm ok the cusparse documentation is confusing. When they say "where op(A) is a sparse matrix with dimensions m × k " I assumed then that when transA = transpose, then op(A) = A^T is mxk which means that the matrix A that you pass to the routine is kxm

Reply:
Ok ill double check that we are doing spmm_csr correctly in rocsparse. Either way I think that this PR is correct as it applies the changes you made to csrmm in rocsparse, to csrmm in hipsparse.